app/backend/src/couchers/jobs/handlers.py

```diff
@@ -221,6 +231,90 @@ def process_send_onboarding_emails(payload):
             session.commit()
 
 
+def process_send_reference_reminders(payload):
+    """
+    Sends out reminders to write references after hosting/staying
+    """
+    logger.info(f"Sending out reference reminder emails")
+
+    # Keep this in chronological order!
+    reference_reminder_schedule = [
+        # (number, days after stay, text for how long they have left to write the ref)
+        # 6 pm ish two days after stay
+        (1, timedelta(days=2) - timedelta(hours=6), "12 days"),
+        # 2 pm ish a week after stay
+        (2, timedelta(days=7) - timedelta(hours=10), "7 days"),
+        # 10 am ish 3 days before end of time to write ref
+        (3, timedelta(days=11) - timedelta(hours=14), "3 days"),
+    ]
+
+    with session_scope() as session:
+        # iterate the reminders in backwards order, so if we missed out on one we don't send duplicates
+        for reminder_no, reminder_time, reminder_text in reversed(reference_reminder_schedule):
+            user = aliased(User)
+            other_user = aliased(User)
+            # surfing reqs
+            q1 = (
+                select(literal(True), HostRequest, user, other_user)
+                .join(user, user.id == HostRequest.from_user_id)
+                .join(other_user, other_user.id == HostRequest.to_user_id)
+                .outerjoin(
+                    Reference,
+                    and_(
+                        Reference.host_request_id == HostRequest.conversation_id,
+                        # if no reference is found, then HostRequest.from_user_id has not written a ref (i.e. the surfer)
+                        Reference.from_user_id == HostRequest.from_user_id,
+                    ),
+                )
+                .where(user.is_visible)
+                .where(other_user.is_visible)
```

Comment:
The normal filters check if one user blocked the other, but since there is no signed in user running background jobs, we can't use those filters.

Also just noticed a bug here: this does not respect blocking.